app/helpers/hosts_helper.rb

```diff
@@ -441,4 +441,23 @@ def lookup_values_data(host_or_hostgroup)
       }
     end
   end
+
+  def virtual?(host)
+    return unless host.reported_data
+
+    host.reported_data.virtual ? _('Yes') : _('No')
+  end
+
+  def humanize_bytes(number, from: nil)
+    return unless number
+
+    number = case from
+             when :mega
+               number * 1_000_000
```

Comment:
This should probably be `number * 1024 * 1024`, or at least for facts coming from ansible.

```
# free -g | head -n 2    
              total        used        free      shared  buff/cache   available
Mem:             31          10           6           1          14          18

# ansible -i ,localhost -c local -m setup localhost | grep ansible_memtotal_mb
        "ansible_memtotal_mb": 31894,
```

With the way it is now, the UI shows `29.7 GB`, with `1024^2` it shows `31.1 GB`.

```suggestion
               number * 1024 * 1024
```

Reply:
Noting here since IRC gets lost, but the source is indeed really MiB. The API is here:
https://github.com/theforeman/foreman/blob/cc7af718235d7d228b94a949c3822284182e50e5/app/services/fact_parser.rb#L112-L114

And at least for Puppet it's normalized here (which clearly shows the same thing):
https://github.com/theforeman/foreman/blob/cc7af718235d7d228b94a949c3822284182e50e5/app/services/puppet_fact_parser.rb#L131-L138

Reply:
Updated